src/main/java/jdbc/web/filter/UserFilter.java

```diff
@@ -24,33 +32,39 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
         HttpServletRequest req = (HttpServletRequest) servletRequest;
         HttpServletResponse res = (HttpServletResponse) servletResponse;
         String path = req.getServletPath() + req.getPathInfo();
-        
-        if (req.getSession().getAttribute("LOGGED_USER") != null) {
-            req.getRequestDispatcher("/WEB-INF/views/home.jsp")
-                    .forward(req, res);
-            return;
-        }
 
         Cookie[] cookies = req.getCookies();
         String token = null;
-        for (Cookie c : cookies) {
-            if (c.getName().equals("MATE")) {
-                token = c.getValue();
+        if (cookies != null) {
+            for (Cookie c : cookies) {
+                if (c.getName().equals("MATE")) {
+                    token = c.getValue();
+                }
             }
         }
 
         if (token == null) {
-            if (path.equals("/servlet/login") || path.equals("/servlet/register")) {
+            if (path.equals("/servlet/login")
+                    || path.equals("/servlet/register")
+                    || path.equals("/servlet/home")
+                    || path.equals("/servlet/user")
```

Comment:
@Dimitriy1 If tokenis not present - than user can visit **"/servlet/home"** or **"/servlet/user"** ?

Reply:
@boroda4436 no I can't visit these pages

Reply:
@BondarenkoJek Do you agree with @Dimitriy1 ?
I am a bit confused, because based on this implementation if token == null then we are calling _processAuthorized_. 
I am ok that "/servlet/login" and "/servlet/register" should be here, but not /home, /user, etc.

@Dimitriy1 
> no I can't visit these pages
In this case where the access should be denied for **"/servlet/home"** or **"/servlet/user"** ?

Reply:
@boroda4436 no, @Dimitriy1 should change this logic
if token = 0 you can open only login or registration page